Failed to build on macOS Montery

**Describe this problem**

Build dependency librocksdb failed, compiler complains `unused-but-set-variable`
```
  /Users/lei/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:105:14: error: variable 'file_size' set but not used [-Werror,-Wunused-but-set-variable]
      uint64_t file_size = 0;
               ^
  1 error generated.
  make[3]: *** [CMakeFiles/titan.dir/src/db_impl.cc.o] Error 1
  make[3]: *** Waiting for unfinished jobs....
  make[2]: *** [CMakeFiles/titan.dir/all] Error 2
  make[1]: *** [CMakeFiles/titan.dir/rule] Error 2
  make: *** [titan] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2

  build script failed, must exit now', /Users/lei/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.48/src/lib.rs:975:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
```

C/CXX version
```
$ cc -v
Apple clang version 13.1.6 (clang-1316.0.21.2.5)
Target: arm64-apple-darwin21.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Rust toolchain version
```
$ rustc --version
rustc 1.59.0-nightly (f1ce0e6a0 2022-01-05)
```

System version

```
$ sw_vers
ProductName:	macOS
ProductVersion:	12.4
BuildVersion:	21F7
```

**Steps to reproduce**

1. Clone repo
2. Install required dependencies including ssl/cmake
3. cargo build

**Expected behavior**

Successfully build binaries.


**Additional Information**

--- a/wal/Cargo.toml
+++ b/wal/Cargo.toml
@@ -20,5 +20,5 @@ futures = { version = "0.3", features = ["async-await"] }
 
 [dependencies.rocksdb]
 git = "https://github.com/tikv/rust-rocksdb.git"
-branch = "tikv-5.2"
+rev = "773784178a0e8e5fdad81f4fd85448a3014a3700"
 features = ["portable"]

